File: producer/jms-producer/src/main/java/com/example/user/adapter/in/UserController.java
package com.example.user.adapter.in;

import com.example.user.domain.User;
import com.example.user.port.in.RegisterUser;

import javax.inject.Inject;
import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.core.MediaType;

@Path("/user")
public class UserController {

    @Inject
    RegisterUser registerUser;

    @POST
    @Path("/save")
    @Consumes(MediaType.APPLICATION_JSON)
    public void saveUser(User user) {
        registerUser.registerUser(user);
    }

}

File: producer/spring-kafka-producer/src/main/kotlin/com/example/springkafkaproducer/user/adapter/in/UserController.kt
package com.example.springkafkaproducer.user.adapter.`in`

import com.example.springkafkaproducer.user.domain.User
import com.example.springkafkaproducer.user.port.`in`.RegisterUser
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/users")
class UserController(private val registerUser: RegisterUser) {

    @PostMapping("/save")
    fun saveUser(@RequestBody user: User) {
        registerUser.registerUser(user)
    }

}
